src/features/recovery/RecoveryDialogs.tsx

```diff
@@ -0,0 +1,381 @@
+}: {
+    orphanedSessions: Session[];
+    onRecoverClick: (session: Session) => void;
+    onRemoveClick: (session: Session) => void;
+}) => (
+    <div className="tw-flex tw-flex-col tw-gap-2">
+        {orphanedSessions.map(session => (
+            <SessionItem
+                key={session.filePath}
+                session={session}
+                onRecoverClick={onRecoverClick}
+                onRemoveClick={onRemoveClick}
+            />
+        ))}
+    </div>
+);
+
+export default () => {
+    const dispatch = useDispatch();
+    const recoveryManager = RecoveryManager.getInstance();
+
+    const pendingRecovery = useSelector(isSessionRecoveryPending);
+    const [isSessionsListDialogVisible, setIsSessionsListDialogVisible] =
+        useState(false);
+    const [orphanedSessions, setOrphanedSessions] = useState<Session[]>([]);
+    const [isRecovering, setIsRecovering] = useState(false);
+    const [recoveryProgress, setRecoveryProgress] = useState(0);
+    const [recoveryError, setRecoveryError] = useState<string | null>(null);
+
+    const lastProgress = useRef(-1);
+    const { time, reset, pause, start } = useStopwatch({
+        autoStart: true,
+        resolution: 1000,
+    });
+
+    useEffect(() => {
+        if (!isRecovering) {
+            lastProgress.current = -1;
+            pause();
+        } else {
+            start(0);
+        }
+    }, [isRecovering, pause, start]);
+
+    useEffect(() => {
+        if (recoveryProgress < lastProgress.current) {
+            lastProgress.current = recoveryProgress;
+            reset();
+        }
+    }, [recoveryProgress, reset]);
+
+    useEffect(() => {
+        if (pendingRecovery) {
+            dispatch(
+                addConfirmBeforeClose({
+                    id: 'sessionRecoveryInProgress',
+                    message:
+                        'There is a session recovery in progress. If you close the application, the progress will be lost and the session will remain in the recovery list. Are you sure you want to close?',
+                })
+            );
+        } else {
+            dispatch(clearConfirmBeforeClose('sessionRecoveryInProgress'));
+        }
+    }, [dispatch, pendingRecovery]);
+
+    useEffect(() => {
+        recoveryManager.searchOrphanedSessions(
+            () => {},
+            (orphanSessions: Session[]) => {
+                if (orphanSessions.length > 0) {
+                    setOrphanedSessions(orphanSessions);
+                    setIsSessionsListDialogVisible(true);
+                }
+            }
+        );
+    }, [recoveryManager]);
+
+    const [showDeleteAllConfirmation, setShowDeleteAllConfirmation] =
+        useState(false);
+
+    return (
+        <>
+            <GenericDialog
+                className="tw-preflight tw-max-h-screen"
+                title={
+                    showDeleteAllConfirmation
+                        ? 'Delete all sessions'
+                        : 'Session Recovery'
```

Comment:
```suggestion
                        : 'Session recovery'
```
Change all occurrences, please :)